tests/Propulsion.Tests/AppendsEpochTests.fs

```diff
@@ -25,5 +26,43 @@ let ``residual span shouldn't be affected by earlier events in closed spans`` ()
 
     test <@ residual1 = spans3 && accepted2 = (spans3 |> Array.map (fun {p = p} -> p)) && residual2 = [||] @>
 
+[<Fact>]
+let ``Already ingested events should be removed by ingestion on closed epoch`` () =
+    let shouldClose = (<) 10
+    let spans1: Events.StreamSpan array = [| { p = IndexStreamId.ofP "Cat-Id"; i = 0L; c = [| "a"; "a" |] } |]
+    let spans2: Events.StreamSpan array = [| { p = IndexStreamId.ofP "Cat-Id"; i = 1L; c = [| "a"; "b" |] } |]
+
+    let (_, events1) = Ingest.decide shouldClose spans1 Fold.initial
+    let epoch1Closed = (Fold.fold Fold.initial events1).WithClosed()
+
+    let ({ accepted = accepted; residual = residual }: ExactlyOnceIngester.IngestResult<_ ,_>, events) = Ingest.decide shouldClose spans2 epoch1Closed
+
+    test <@ accepted = [||] && residual = [| { p = IndexStreamId.ofP "Cat-Id"; i = 2L; c = [| "b" |] } |]@>
+
+[<Fact>]
+let ``Already ingested events are not ingested on open epoch`` () =
+    let streamId = IndexStreamId.ofP "Cat-Id"
+    let shouldClose = (<) 10
+    let spans1: Events.StreamSpan array = [| { p = streamId; i = 0L; c = [| "a"; "a" |] } |]
```

Comment:
pull out a helper
```
let mkSpan sid index cases: Events.StreamSpan = { p = IndexStreamId.ofP sid; i = index; c = cases }
```
Then this can become
```suggestion
    let spans1 = [| mkSpan "Cat-Id" 0L [| "a"; "a" |] |]
```
or

```
    let spans1 = mkSpan "Cat-Id" 0L [| "a"; "a" |] |> Array.singleton
```